app/src/main/java/org/kiwix/kiwixmobile/utils/LanguageUtils.java

```diff
@@ -341,16 +341,20 @@ public String getLanguageName() {
     }
   }
 
-  public static String getStringOrResource(Context appContext, String str){
-    if(str.contains("REPLACE_")) {
-      int resourceId = appContext.getResources()
-              .getIdentifier(
-                      str.replace("REPLACE_", ""),
-                      "string",
-                      appContext.getPackageName()
-              );
-      return appContext.getResources().getString(resourceId);
+
+
+  public static String getResourceString(Context appContext, String str){
+    String resourceName = str;
+    if(resourceName.contains("REPLACE_")) {
+      resourceName = resourceName.replace("REPLACE_", "");
```

Comment:
What if the resourceName doesn't contain REPLACE_ ... what should the rest of this method do? presumably it should return rather than continuing to try and find a matching resource?

Reply:
This method now accepts `str` as a `REPLACE_` string or a normal resource name
Possibly the replace should be moved to the caller of the function?